app/models/spree/relation.rb

```diff
@@ -6,4 +6,32 @@ class Spree::Relation < ApplicationRecord
   belongs_to :related_to, polymorphic: true
 
   validates :relation_type, :relatable, :related_to, presence: true
+
+  after_create :create_inverse, unless: :has_inverse?, if: :bidirectional?
+  after_destroy :destroy_inverses,
+    if: Proc.new { |relation| relation.bidirectional? && relation.has_inverse? }
+
+  delegate :bidirectional?, to: :relation_type
+
+  def has_inverse?
+    self.class.exists?(inverse_options)
+  end
+
+  def inverses
+    self.class.where(inverse_options)
+  end
+
+  def inverse_options
+    { relation_type: relation_type, relatable: related_to, related_to: relatable }
+  end
+
+  private
+
+  def create_inverse
+    self.class.create(inverse_options.merge(description: description))
+  end
+
+  def destroy_inverses
+    inverses.destroy_all
```

Comment:
Same here... Should we call `.each(&:destroy!)` instead so that it fails loudly?